modules/client/shared/src/main/scala/com/azavea/stac4s/api/client/BaseSttpStacClient.scala

```diff
@@ -0,0 +1,105 @@
+package com.azavea.stac4s.api.client
+
+import com.azavea.stac4s.{StacCollection, StacItem}
+
+import cats.MonadError
+import cats.syntax.flatMap._
+import cats.syntax.functor._
+import cats.syntax.option._
+import eu.timepit.refined.types.string.NonEmptyString
+import io.circe.syntax._
+import io.circe.{Encoder, Json}
+import sttp.client3.circe.asJson
+import sttp.client3.{SttpBackend, basicRequest}
+import sttp.model.Uri
+
+abstract class BaseSttpStacClient[F[_]: MonadError[*[_], Throwable]](
+    client: SttpBackend[F, Any],
+    baseUri: Uri
+) extends StacClient[F] {
+
+  type Filter
```

Comment:
I'm a little confused -- what's the purpose of a dependent `Filter` type?

Reply:
@jisantuc the trick here is to have it in the `shared` package. To put `STACClient` into the `shared` package we only need to have `SearchFilters` compilinf for both JS and JVM backends. 

`SearchFilters` should countain `jts.Geometry` type that is why we need to maintain a spearate implementaion for the JS backend. 

Having this `Filter` type hidden helps to simplify STACClient signature. 

`StacClient[F]` signatures makes sense but `StacClient[F, SearchFilter]` looks extremely confusing.

The alternative to that can be found here 0647db7

Reply:
Btw we can have a quick call if you want me to walk you through this decision